### ISSUE TITLE
bun repl crashes

### ISSUE THREAD

**Issue by frm-037** [oven-sh/bun#12107]
### How can we reproduce the crash?

Is very easy to reproduce:

- Install 1.1.15
- Install bun repl
- upgrade to 1.1.16
- Run bun repl -> crash

### Relevant log output

```shell
$ bun repl
============================================================
Bun v1.1.16 (bf7b327f) Linux x64
Args: "bun" "bun-repl"
Elapsed: 1ms | User: 0ms | Sys: 7ms
RSS: 1.07GB | Peak: 9.47MB | Commit: 1.07GB | Faults: 0

panic(main thread): Segmentation fault at address 0x610EFF37972F
oh no: Bun has crashed. This indicates a bug in Bun, not your code.
```

### Stack Trace (bun.report)

Bun v1.1.16 ([`bf7b327`](<https://github.com/oven-sh/bun/tree/bf7b327f68568104a5e40c1944b7b255b02f0745>)) on linux x86_64 [BunxCommand]

Segmentation fault at address 0x56C0A9C9372F

- `??`
- [`semver.zig:268`](<https://github.com/oven-sh/bun/blob/bf7b327f68568104a5e40c1944b7b255b02f0745/src/install/semver.zig#L268>): `src.cli.bunx_command.BunxCommand.exec`
- [`StandaloneModuleGraph.zig:677`](<https://github.com/oven-sh/bun/blob/bf7b327f68568104a5e40c1944b7b255b02f0745/src/StandaloneModuleGraph.zig#L677>): `src.cli.Cli.start`
- [`array_list.zig:0`](<https://github.com/oven-sh/bun/blob/bf7b327f68568104a5e40c1944b7b255b02f0745/src/deps/zig/lib/std/array_list.zig#L0>): `process.argsAlloc`

**Comment by Jarred-Sumner** [comment]
Duplicate of #12070

**Comment by Jarred-Sumner** [comment]
Fixed in #12071 

Run `bun upgrade --canary` to get the fix before Bun v1.1.17
